odoo/tools/misc.py

```diff
@@ -1013,19 +1013,34 @@ def popmap(self):
 class OrderedSet(MutableSet):
     """ A set collection that remembers the elements first insertion order. """
     __slots__ = ['_map']
+
     def __init__(self, elems=()):
         self._map = OrderedDict((elem, None) for elem in elems)
+
     def __contains__(self, elem):
         return elem in self._map
+
     def __iter__(self):
         return iter(self._map)
+
     def __len__(self):
         return len(self._map)
+
     def add(self, elem):
         self._map[elem] = None
+
     def discard(self, elem):
         self._map.pop(elem, None)
 
+    def update(self, elems):
+        for elem in elems:
+            self.add(elem)
```

Comment:
Is this called a lot and / or with lots of items? Might be worth looking at
```python
self._map.update(zip(elems, itertools.repeat(None)))
```
or
```python
self._map.update(zip(elems, [None]*len(elems)))
```
if it is

Reply:
I add an optionnal commit using this suggestion and replacing OrderedDict by dict. We can still  discuss if it is acceptable/needed.